ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/MachineManager.java

```diff
@@ -21,6 +23,26 @@
 public interface MachineManager {
 
     /**
+     * Get machine information by it's id.
+     *
+     * @param workspaceId
+     *         ID of workspace
+     * @param machineId
+     *         ID of the machine
+     * @return a promise that resolves to the {@link MachineEntity}, or rejects with an error
+     */
+    Promise<MachineEntity> getMachine(final String workspaceId, final String machineId);
+
+    /**
+     * Returns list of machines which are bounded to the specified workspace.
```

Comment:
**"bouded/unbounded machine"** - it's old obsolete terminology.
IMO **"Returns list of all machines which of the specified workspace"** would be better.